### PR TITLE
Check TTL before creating index

### PR DESCRIPTION
**Pull request by panda-sheep**
```
It mainly includes the following parts
1)    use ttl when create tag/edge,  drop ttl when alter tag/edge
       not use ttl when create tag/edge ,  add ttl when alter tag/edge
2)     tag/edge contains index,   not  used  ttl when alter tag/edge
3)     tag/edge uses ttl,   not  create index
4)     add the above unit tests

5)    change columns are not allowed when the column is ttl_col
```

### CODE REVIEW

File: src/meta/MetaServiceUtils.cpp [hunk 1]
```diff
@@ -423,15 +423,15 @@ cpp2::ErrorCode MetaServiceUtils::alterColumnDefs(std::vector<nebula::cpp2::Colu
...
-                        (prop.get_ttl_col() && (*prop.get_ttl_col() != col.get_name()))) {
-                        cols.erase(it);
-                        return cpp2::ErrorCode::SUCCEEDED;
-                    } else {
-                        LOG(ERROR) << "Column can't be dropped, a TTL attribute on it: "
-                                   << col.get_name();
-                        return cpp2::ErrorCode::E_NOT_DROP;
+                    // drop the column with ttl
+                    // if (!prop.get_ttl_col() ||
+                    //    (prop.get_ttl_col() && (*prop.get_ttl_col() != col.get_name()))) {
```
Comment (bright-starry-sky) [inline src/meta/MetaServiceUtils.cpp]: Extra comments.

Comment (panda-sheep) [reply src/meta/MetaServiceUtils.cpp]: Yeah, good catch

File: src/meta/processors/BaseProcessor.h [hunk 2]
```diff
@@ -226,12 +233,21 @@ class BaseProcessor {
...
+     * Fetch the latest version edge's schema.
+     */
+    StatusOr<nebula::cpp2::Schema>
+    getLatestEdgeSchema(GraphSpaceID spaceId, const EdgeType edgeType);
+
+    /**
+     * Check if edge has ttl
+     */
+    StatusOr<bool>
+    edgeHasTTL(GraphSpaceID spaceId, const EdgeType edgeType);
```
Comment (dangleptr) [inline src/meta/processors/BaseProcessor.h]: Do we need the method?

Comment (panda-sheep) [reply src/meta/processors/BaseProcessor.h]: This function is deprecated. I remember it has been removed.

**Comment by panda-sheep** [comment]
The regression test issue has been fixed（#1779）, please ignore it here

**Pull request merged**